packages/react/src/utils/floating-ui/useClick.ts

```diff
@@ -100,12 +100,13 @@ export function useClick(context: FloatingRootContext, props: UseClickProps = {}
           return;
         }
 
+        const openEvent = dataRef.current.openEvent;
+
         const nextOpen = !(
           open &&
           toggle &&
-          (dataRef.current.openEvent && stickIfOpen
-            ? dataRef.current.openEvent.type === 'click' ||
-              dataRef.current.openEvent.type === 'mousedown'
+          (openEvent && stickIfOpen
+            ? ['click', 'mousedown', 'keydown', 'keyup'].includes(openEvent.type)
```

Comment:
I think Romain would prefer to avoid arrays here :laughing: 
To keep the lise of `||` as before, or at least to have a `Set` outside of the function to avoid `O(n)`

Reply:
Yeah I knew 😄  - I just figured since it's a discrete click event that fires infrequently, the perf concerns are not an issue

Reply:
I don't think there will be any significant difference for just 4 elements. One thing I'd suggest, though is to have a constant array, not define it every time the handler runs.